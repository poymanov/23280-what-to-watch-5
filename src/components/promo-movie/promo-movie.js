import React, {Fragment} from "react";
import MovieTypes from "../../types/movies";
import PropTypes from "prop-types";
import {connect} from "react-redux";

const PromoMovie = ({promo, onPlayButtonClick}) => {
  return (
    <Fragment>
      <div className="movie-card__bg" style={{backgroundColor: promo.backgroundColor}}>
        <img src={promo.backgroundImage} alt={promo.name}/>
      </div>

      <div className="movie-card__wrap">
        <div className="movie-card__info">
          <div className="movie-card__poster">
            <img src={promo.posterImage} alt={promo.name + ` poster`} width="218" height="327"/>
          </div>

          <div className="movie-card__desc">
            <h2 className="movie-card__title">{promo.name}</h2>
            <p className="movie-card__meta">
              <span className="movie-card__genre">{promo.genre}</span>
              <span className="movie-card__year">{promo.released}</span>
            </p>

            <div className="movie-card__buttons">
              <button className="btn btn--play movie-card__button" type="button" onClick={() => onPlayButtonClick(1)}>
                <svg viewBox="0 0 19 19" width="19" height="19">
                  <use href="#play-s" />
                </svg>
                <span>Play</span>
              </button>
              <button className="btn btn--list movie-card__button" type="button">
                <svg viewBox="0 0 19 20" width="19" height="20">
                  <use href="#add" />
                </svg>
                <span>My list</span>
              </button>
            </div>
          </div>
        </div>
      </div>
    </Fragment>
  );
};

PromoMovie.propTypes = {
  promo: MovieTypes.promo,
  onPlayButtonClick: PropTypes.func.isRequired
};

const mapStateToProps = ({MOVIES}) => ({
  promo: MOVIES.promo,
});

export {PromoMovie};
export default connect(mapStateToProps)(PromoMovie);
