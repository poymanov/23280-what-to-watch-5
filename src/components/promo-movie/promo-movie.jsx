import React, {Fragment, PureComponent} from "react";
import MovieTypes from "../../types/movies";
import PropTypes from "prop-types";
import {connect} from "react-redux";
import {fetchPromoMovie, addMovieToFavorite} from "../../store/api-actions";
import {isUserAuthSelector, promoMovieSelector} from "../../store/selectors";

class PromoMovie extends PureComponent {
  constructor(props) {
    super(props);
  }

  componentDidMount() {
    this.props.fetchPromoMovie();
  }

  render() {
    const {promo, onPlayButtonClick, isUserAuth, addToFavorite} = this.props;

    if (!promo) {
      return null;
    }

    let addToListButton = null;

    if (isUserAuth) {
      addToListButton = <button className="btn btn--list movie-card__button" type="button" onClick={() => addToFavorite(promo.id)}>
        <svg viewBox="0 0 19 20" width="19" height="20">
          <use href="#add"/>
        </svg>
        <span>My list</span>
      </button>;
    }

    return (
      <Fragment>
        <div className="movie-card__bg" style={{backgroundColor: promo.backgroundColor}}>
          <img src={promo.backgroundImage} alt={promo.name}/>
        </div>

        <div className="movie-card__wrap">
          <div className="movie-card__info">
            <div className="movie-card__poster">
              <img src={promo.posterImage} alt={promo.name + ` poster`} width="218" height="327"/>
            </div>

            <div className="movie-card__desc">
              <h2 className="movie-card__title">{promo.name}</h2>
              <p className="movie-card__meta">
                <span className="movie-card__genre">{promo.genre}</span>
                <span className="movie-card__year">{promo.released}</span>
              </p>

              <div className="movie-card__buttons">
                <button className="btn btn--play movie-card__button" type="button" onClick={() => onPlayButtonClick(1)}>
                  <svg viewBox="0 0 19 19" width="19" height="19">
                    <use href="#play-s"/>
                  </svg>
                  <span>Play</span>
                </button>
                {addToListButton}
              </div>
            </div>
          </div>
        </div>
      </Fragment>
    );
  }
}

PromoMovie.propTypes = {
  promo: MovieTypes.promo,
  onPlayButtonClick: PropTypes.func.isRequired,
  fetchPromoMovie: PropTypes.func.isRequired,
  isUserAuth: PropTypes.bool.isRequired,
  addToFavorite: PropTypes.func.isRequired
};

const mapStateToProps = (state) => ({
  promo: promoMovieSelector(state),
  isUserAuth: isUserAuthSelector(state),
});

const mapDispatchToProps = (dispatch) => ({
  fetchPromoMovie() {
    dispatch(fetchPromoMovie());
  },
  addToFavorite(movieId) {
    dispatch(addMovieToFavorite(movieId));
  }
});

export {PromoMovie};
export default connect(mapStateToProps, mapDispatchToProps)(PromoMovie);
