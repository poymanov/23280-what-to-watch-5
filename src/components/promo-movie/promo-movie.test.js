import React from "react";
import renderer from "react-test-renderer";
import {PromoMovie} from "./promo-movie";

const noop = () => {};

const movie = {
  id: 1,
  name: `test`,
  posterImage: `test`,
  previewImage: `test`,
  backgroundImage: `test`,
  backgroundColor: `test`,
  videoLink: `test`,
  previewVideoLink: `test`,
  description: `test`,
  rating: 1,
  scoresCount: 1,
  director: `test`,
  starring: [`test`],
  runTime: 1,
  genre: `test`,
  released: 1,
  isFavorite: false,
};

describe(`Should PromoMovie render correctly`, () => {
  it(`Without movie`, () => {
    const tree = renderer
      .create(<PromoMovie isUserAuth={false} addToFavorite={noop} fetchPromoMovie={noop} onPlayButtonClick={noop}/>)
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it(`With movie`, () => {
    const tree = renderer
      .create(<PromoMovie isUserAuth={false} addToFavorite={noop} promo={movie} fetchPromoMovie={noop} onPlayButtonClick={noop}/>)
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it(`With auth user`, () => {
    const tree = renderer
      .create(<PromoMovie isUserAuth={true} addToFavorite={noop} promo={movie} fetchPromoMovie={noop} onPlayButtonClick={noop}/>)
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});


