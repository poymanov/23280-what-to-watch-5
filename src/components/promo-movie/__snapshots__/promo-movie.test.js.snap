// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should PromoMovie render correctly With auth user 1`] = `
Array [
  <div
    className="movie-card__bg"
    style={
      Object {
        "backgroundColor": "test",
      }
    }
  >
    <img
      alt="test"
      src="test"
    />
  </div>,
  <div
    className="movie-card__wrap"
  >
    <div
      className="movie-card__info"
    >
      <div
        className="movie-card__poster"
      >
        <img
          alt="test poster"
          height="327"
          src="test"
          width="218"
        />
      </div>
      <div
        className="movie-card__desc"
      >
        <h2
          className="movie-card__title"
        >
          test
        </h2>
        <p
          className="movie-card__meta"
        >
          <span
            className="movie-card__genre"
          >
            test
          </span>
          <span
            className="movie-card__year"
          >
            1
          </span>
        </p>
        <div
          className="movie-card__buttons"
        >
          <button
            className="btn btn--play movie-card__button"
            onClick={[Function]}
            type="button"
          >
            <svg
              height="19"
              viewBox="0 0 19 19"
              width="19"
            >
              <use
                href="#play-s"
              />
            </svg>
            <span>
              Play
            </span>
          </button>
          <button
            className="btn btn--list movie-card__button"
            onClick={[Function]}
            type="button"
          >
            <svg
              height="20"
              viewBox="0 0 19 20"
              width="19"
            >
              <use
                href="#add"
              />
            </svg>
            <span>
              My list
            </span>
          </button>
        </div>
      </div>
    </div>
  </div>,
]
`;

exports[`Should PromoMovie render correctly With auth user and favorited movie 1`] = `
Array [
  <div
    className="movie-card__bg"
    style={
      Object {
        "backgroundColor": "test",
      }
    }
  >
    <img
      alt="test"
      src="test"
    />
  </div>,
  <div
    className="movie-card__wrap"
  >
    <div
      className="movie-card__info"
    >
      <div
        className="movie-card__poster"
      >
        <img
          alt="test poster"
          height="327"
          src="test"
          width="218"
        />
      </div>
      <div
        className="movie-card__desc"
      >
        <h2
          className="movie-card__title"
        >
          test
        </h2>
        <p
          className="movie-card__meta"
        >
          <span
            className="movie-card__genre"
          >
            test
          </span>
          <span
            className="movie-card__year"
          >
            1
          </span>
        </p>
        <div
          className="movie-card__buttons"
        >
          <button
            className="btn btn--play movie-card__button"
            onClick={[Function]}
            type="button"
          >
            <svg
              height="19"
              viewBox="0 0 19 19"
              width="19"
            >
              <use
                href="#play-s"
              />
            </svg>
            <span>
              Play
            </span>
          </button>
          <button
            className="btn btn--list movie-card__button"
            onClick={[Function]}
            type="button"
          >
            <svg
              height="20"
              viewBox="0 0 19 20"
              width="19"
            >
              <use
                href="#in-list"
              />
            </svg>
            <span>
              My list
            </span>
          </button>
        </div>
      </div>
    </div>
  </div>,
]
`;

exports[`Should PromoMovie render correctly With movie 1`] = `
Array [
  <div
    className="movie-card__bg"
    style={
      Object {
        "backgroundColor": "test",
      }
    }
  >
    <img
      alt="test"
      src="test"
    />
  </div>,
  <div
    className="movie-card__wrap"
  >
    <div
      className="movie-card__info"
    >
      <div
        className="movie-card__poster"
      >
        <img
          alt="test poster"
          height="327"
          src="test"
          width="218"
        />
      </div>
      <div
        className="movie-card__desc"
      >
        <h2
          className="movie-card__title"
        >
          test
        </h2>
        <p
          className="movie-card__meta"
        >
          <span
            className="movie-card__genre"
          >
            test
          </span>
          <span
            className="movie-card__year"
          >
            1
          </span>
        </p>
        <div
          className="movie-card__buttons"
        >
          <button
            className="btn btn--play movie-card__button"
            onClick={[Function]}
            type="button"
          >
            <svg
              height="19"
              viewBox="0 0 19 19"
              width="19"
            >
              <use
                href="#play-s"
              />
            </svg>
            <span>
              Play
            </span>
          </button>
        </div>
      </div>
    </div>
  </div>,
]
`;

exports[`Should PromoMovie render correctly Without movie 1`] = `null`;
